File: win-prog/ProgWinCS/Taming the Mouse/MouseCursorsProperty/MouseCursorsProperty.cs
//---------------------------------------------------
// MouseCursorsProperty.cs © 2001 by Charles Petzold
//---------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class MouseCursorsProperty: Form
{
     Label[] acntl = new Label[28];

     public static void Main()
     {
          Application.Run(new MouseCursorsProperty());
     }
     public MouseCursorsProperty()
     {
          Cursor[] acursor = 
          { 
               Cursors.AppStarting, Cursors.Arrow,       Cursors.Cross,       
               Cursors.Default,     Cursors.Hand,        Cursors.Help,     
               Cursors.HSplit,      Cursors.IBeam,       Cursors.No,          
               Cursors.NoMove2D,    Cursors.NoMoveHoriz, Cursors.NoMoveVert,
               Cursors.PanEast,     Cursors.PanNE,       Cursors.PanNorth,    
               Cursors.PanNW,       Cursors.PanSE,       Cursors.PanSouth,
               Cursors.PanSW,       Cursors.PanWest,     Cursors.SizeAll,     
               Cursors.SizeNESW,    Cursors.SizeNS,      Cursors.SizeNWSE,
               Cursors.SizeWE,      Cursors.UpArrow,     Cursors.VSplit,      
               Cursors.WaitCursor
          };
          string[] astrCursor = 
          { 
               "AppStarting",       "Arrow",             "Cross",       
               "Default",           "Hand",              "Help",     
               "HSplit",            "IBeam",             "No",          
               "NoMove2D",          "NoMoveHoriz",       "NoMoveVert",
               "PanEast",           "PanNE",             "PanNorth",    
               "PanNW",             "PanSE",             "PanSouth",
               "PanSW",             "PanWest",           "SizeAll",     
               "SizeNESW",          "SizeNS",            "SizeNWSE",
               "SizeWE",            "UpArrow",           "VSplit",      
               "WaitCursor" 
          };

          Text = "Mouse Cursors Using Cursor Property";

          for (int i = 0; i < 28; i++)
          {
               acntl[i] = new Label();
               acntl[i].Parent = this;
               acntl[i].Text = astrCursor[i];
               acntl[i].Cursor = acursor[i]; 
               acntl[i].BorderStyle = BorderStyle.FixedSingle;
          }
		 OnResize(EventArgs.Empty);
     }
     protected override void OnResize(EventArgs ea)
     {
          for (int i = 0; i < acntl.Length; i++)
          {
               acntl[i].Bounds = Rectangle.FromLTRB(
                                        (i % 4    ) * ClientSize.Width  / 4, 
                                        (i / 4    ) * ClientSize.Height / 7,
                                        (i % 4 + 1) * ClientSize.Width  / 4, 
                                        (i / 4 + 1) * ClientSize.Height / 7);
          }
     }
}

File: win-prog/ProgWinVB/Taming the Mouse/MouseCursorsProperty/MouseCursorsProperty.vb
'-----------------------------------------------------
' MouseCursorsProperty.vb (c) 2002 by Charles Petzold
'-----------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class MouseCursorsProperty
    Inherits Form

    Private actrl(28) As Label

    Shared Sub Main()
        Application.Run(New MouseCursorsProperty())
    End Sub

    Sub New()
        Dim i As Integer
        Dim acursor() As Cursor = _
        { _
            Cursors.AppStarting, Cursors.Arrow, Cursors.Cross, _
            Cursors.Default, Cursors.Hand, Cursors.Help, _
            Cursors.HSplit, Cursors.IBeam, Cursors.No, _
            Cursors.NoMove2D, Cursors.NoMoveHoriz, Cursors.NoMoveVert, _
            Cursors.PanEast, Cursors.PanNE, Cursors.PanNorth, _
            Cursors.PanNW, Cursors.PanSE, Cursors.PanSouth, _
            Cursors.PanSW, Cursors.PanWest, Cursors.SizeAll, _
            Cursors.SizeNESW, Cursors.SizeNS, Cursors.SizeNWSE, _
            Cursors.SizeWE, Cursors.UpArrow, Cursors.VSplit, _
            Cursors.WaitCursor _
        }
        Dim astrCursor() As String = _
        { _
            "AppStarting", "Arrow", "Cross", "Default", "Hand", _
            "Help", "HSplit", "IBeam", "No", "NoMove2D", _
            "NoMoveHoriz", "NoMoveVert", "PanEast", "PanNE", "PanNorth", _
            "PanNW", "PanSE", "PanSouth", "PanSW", "PanWest", _
            "SizeAll", "SizeNESW", "SizeNS", "SizeNWSE", "SizeWE", _
            "UpArrow", "VSplit", "WaitCursor" _
        }
        Text = "Mouse Cursors Using Cursor Property"

        For i = 0 To 27
            actrl(i) = New Label()
            actrl(i).Parent = Me
            actrl(i).Text = astrCursor(i)
            actrl(i).Cursor = acursor(i)
            actrl(i).BorderStyle = BorderStyle.FixedSingle
        Next i
        OnResize(EventArgs.Empty)
    End Sub

    Protected Overrides Sub OnResize(ByVal ea As EventArgs)
        Dim i As Integer
        For i = 0 To 27
            actrl(i).Bounds = Rectangle.FromLTRB( _
                                (i Mod 4) * ClientSize.Width \ 4, _
                                (i \ 4) * ClientSize.Height \ 7, _
                                (i Mod 4 + 1) * ClientSize.Width \ 4, _
                                (i \ 4 + 1) * ClientSize.Height \ 7)
        Next i
    End Sub
End Class
